### PR TITLE
Logging audit

### PR DESCRIPTION
**Pull request by Pchelolo**
Self requires the message in the log to be under the field 'msg', but we name it incorrectly sometimes. This fixes it.

Bug: https://phabricator.wikimedia.org/T108189

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -129,23 +129,41 @@ Logger.prototype.child = function(args) {
 
 Logger.prototype.log = function(level, info) {
     var levelMatch = this._levelMatcher.exec(level);
-    if (levelMatch) {
+    if (info && levelMatch) {
         var logger = this._logger;
         var simpleLevel = levelMatch[1];
-        if (info && info instanceof String) {
-            info = {msg: info};
```
Comment (Pchelolo) [inline lib/logger.js]: This was a double-bug. `"test" instanceof String` is actually `false`, but we didn't need to wrap a message into an object - it gets overriden by bunyan with a third argument value (or by undefined).

File: lib/logger.js [hunk 2]
```diff
@@ -129,23 +129,41 @@ Logger.prototype.child = function(args) {
...
+        if (typeof info === 'string') {
+            logger[simpleLevel].call(logger, info);
+        } else if (typeof info === 'object') {
+            var msg;
             // Got an object
             //
+            // If there's a wrapped Error - unwrap and send,
+            // bunyan will properly handle it
+            if (info.err instanceof Error) {
+                info = info.err;
```
Comment (Pchelolo) [inline lib/logger.js]: Errors are correctly handled by bunyan, no need to change anything about them

File: service-runner.js [hunk 3]
```diff
@@ -69,13 +69,13 @@ ServiceRunner.prototype.run = function run(conf) {
 
         // Set up the logger
         if (!config.logging.name) {
-            config.logging.name = name;
+            config.logging.name = (config.services && config.services.name) || name;
```
Comment (d00rman) [inline service-runner.js]: `config.services` is an Array. But. this is not the right place nor way to set the name. [`_runWorker()`](https://github.com/wikimedia/service-runner/blob/master/service-runner.js#L278-L285) conveys that info to individual services and already sets the name.

File: service-runner.js [hunk 4]
```diff
@@ -69,13 +69,13 @@ ServiceRunner.prototype.run = function run(conf) {
...
         }
         self._logger = new Logger(config.logging);
 
         // And the statsd client
         if (!config.metrics.name) {
-            config.metrics.name = name;
+            config.metrics.name = (config.services && config.services.name) || name;
```
Comment (d00rman) [inline service-runner.js]: Idem as above, with the distinction that `config.metrics` is a shared object, so a new metrics object should be created for each service (in [`_runWorker()`](https://github.com/wikimedia/service-runner/blob/master/service-runner.js#L278-L285))

**Comment by Pchelolo** [comment]
@d00rman Updated according to your comments.

File: service-runner.js [hunk 5]
```diff
@@ -252,6 +251,8 @@ ServiceRunner.prototype._runWorker = function() {
         process.chdir(cwd);
     });
 
+    self._metrics = makeStatsD(self.config.metrics, self._logger);
```
Comment (d00rman) [inline service-runner.js]: Need to add the check for `self.config.metrics.name` (from L#76), and set it to the value of L#56

**Pull request merged**